### DISCUSSION TITLE
istio-guide/docs/common-problem/upstream-connection-termination/

### DISCUSSION THREAD

**Discussion by giscus** [zhaohuabing/istio-guide#4] [Announcements]
# istio-guide/docs/common-problem/upstream-connection-termination/

Upstream 断开链路导致 503 UC。

https://www.zhaohuabing.com/istio-guide/docs/common-problem/upstream-connection-termination/

**Comment by LugiaChang** [comment]
这里服务器端 upstream 是在集群内吗，还是集群外的？集群内的话 server 会有 istio-proxy 吗？

**Reply by zhaohuabing** [reply]
这里的 503 是由于 upstream 主动断链，但客户端 envoy sidecar 还没有探测到，以至于客户端访问时使用了过期的链接。和服务器端是否在集群内无关。

**Reply by LugiaChang** [reply]
因为我这有个类似的场景也出现了这个问题，是在客户端 envoy 里的 wasm 插件发起 http call，server 端没有istio-proxy时就会马上断开连接，netstat 看就是 TIME_WAIT，然后 wasm 那边偶现 connection termination 错误（不确定和server端马上断开连接有没有关系）。但是如果经过了 istio-proxy，就会一直保持 establish，这样客户端 envoy 还会认为 upstream 断链吗？

**Comment by zhaohuabing** [comment]
我估计服务器端有sidecar的情况下，客户端不会出现该错误，因为客户端的HTTP链接实际上是连到服务器的sidecar上的，而服务器端的sidecar不会主动断开链接。但服务器端sidecar访问pod内的服务端时可能出现该错误。
